boards/raspberrypi/rpi_pico_2/rpi_pico_2-pinctrl.dtsi

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2024, Andrew Featherstone
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/dt-bindings/pinctrl/rpi-pico-rp2040-pinctrl.h>
+
+&pinctrl {
+	uart0_default: uart0_default {
+		group1 {
+			pinmux = <UART0_TX_P0>;
+		};
+		group2 {
+			pinmux = <UART0_RX_P1>;
+			input-enable;
+		};
+	};
+
+	spi0_default: spi0_default {
+		group1 {
+			pinmux = <SPI0_CSN_P17>, <SPI0_SCK_P18>, <SPI0_TX_P19>;
+		};
+		group2 {
+			pinmux = <SPI0_RX_P16>;
+			input-enable;
+		};
+	};
+
+	clocks_default: clocks_default {
+	};
```

Comment:
empty section can go

Reply:
I'd like to keep this as-is. This is copied from existing RP2040-based boards, and removing the empty section breaks boards/raspberrypi/rpi_pico_2/rpi_pico_2-common.dtsi :
```
&clocks {
	pinctrl-0 = <&clocks_default>;
	pinctrl-names = "default";
};
```

If this needs to go, I feel it should be done in a separate PR for all the existing boards, otherwise the scope of this PR becomes a bit too broad. @soburi , it looks like you introduced this in 99a9b995d38 , can you provide a more authoritative answer on this?

Reply:
The rp2350 and rp2040 are different, so it doesn't necessarily have to be the same as the rp2040.
This was too much, so let's eliminate it.

Reply:
This is done on the current `HEAD` of this branch. I'll leave it as an exercise for the reader to remove these from the RP2040-based boards ;-).